tests/jquery.flot.time.Test.js

```diff
@@ -56,6 +56,17 @@ describe('A Flot chart with absolute time axes', function () {
         ]);
     });
 
+    it('creates enough time ticks', function() {
+        plot = createPlotWithAbsoluteTimeAxis(placeholder, [[[0.95, 1], [1.95, 2]]], undefined, 'seconds');
+
+        var ticks = plot.getAxes().xaxis.ticks;
+        expect(ticks.length).toBeGreaterThan(10);
+        ticks.sort((a, b) => a.v - b.v);
+        for (var i = 1; i < ticks.length; i++) {
+            expect(ticks[i].v - ticks[i - 1].v).toBeLessThan(0.5);
```

Comment:
This seems like a strange test, at least at face value. Sorting ticks prior to doing any assertions on the ticks that depend on that order doesn't seem like a good idea (even if the particular test being written works). The question that immediately comes to mind is "Why does one need to sort the ticks before asserting on a condition respective to the values between each successive tick?".

Perhaps another way to write the test to provide the same confidence in the implementation is that you have the expected number of ticks (already present in test), and that the first tick is a particular number, and that the last tick is a particular number. The in-between tick values are less important and we can rely on other tests (hopefully) that that logic is well tested.

Am I misunderstanding a necessary aspect of what is being tested here?

Reply:
Hi again @atmgrifter00 

I'm not a fan of sorting ticks either. Reason for this is/was:

1. Sorting ticks is already made in this same file:
https://github.com/flot/flot/blob/0b84bfb148922c000768064a2391f2ba4a5d97bc/tests/jquery.flot.time.Test.js#L78

2. Avoid using "hard-coded" values in the test suite. I actually hesitated to use the approach you suggested (counting ticks & checking first/last tick values) based on test values. I agree it's testing more shallowly too. So, test would look like this with the other approach:

Test would be:
```javascript
    it('creates enough time ticks', function() {
        plot = createPlotWithAbsoluteTimeAxis(placeholder, [[[0.95, 1], [1.95, 2]]], undefined, 'seconds');

        var ticks = plot.getAxes().xaxis.ticks;
        expect(ticks.length).toEqual(14);
        expect(ticks[0].v).toEqual(0.95);
        expect(ticks[ticks.length - 1].v).toEqual(1.95);
    });
```
I'm happy to change it - Waiting for your green light to push

Reply:
Eh. Hard-coded values aren't anathema to me in tests. They provide the clearest expectation for what the test in question is meant to illustrate. Granted, when writing tests with hard-coded values it can be beneficial to provide multiple varied sets of input values.

Another important aspect of the change you are making that really isn't captured in the test, is the fact that the ticks are distributed across the length of the plot area, instead of more-or-less one half of it, which is the real behavior of interest (not the amount of ticks that were generated). It's unclear to me what the best way to test this would be. It might be sufficient to check that the min tick is less than 1/4 of the way along the plot area width, and a similar check for the max tick (for a given size of the plot area).

Reply:
Yea, to capture this, sorting the ticks and checking that there's no "bigger gap than a certain value, given the size of plot area, between each two consecutive ticks" was my original idea - it has the disadvantage of testing more than what the original bug was causing:
```javascript
        ticks.sort((a, b) => a.v - b.v);
        for (var i = 1; i < ticks.length; i++) {
            expect(ticks[i].v - ticks[i - 1].v).toBeLessThan(0.5);
        }
```

For reference flot generates these ticks for this test-case:
- Before the fix: `[0.95, 1.8, 1.9, 2, 1.95]`
- After fix: `[0.95, 0.9, 1, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2, 1.95]`

Given our discussion about sorting ticks and your suggestion, this test could be
```javascript
    it('creates enough time ticks', function() {
        plot = createPlotWithAbsoluteTimeAxis(placeholder, [[[0.95, 1], [1.95, 2]]], undefined, 'seconds');

        var ticks = plot.getAxes().xaxis.ticks;
        expect(ticks.length).toEqual(14);
        expect(ticks[0].v).toEqual(0.95);
        expect(ticks[ticks.length - 1].v).toEqual(1.95);
        var minTickValue = Math.min(...ticks.map(tick => tick.v));
        expect(minTickValue).toBeLessThan(1.45);
    });
```

1.2 being min value + 25% of plot width. I don't think it's necessary to test the max tick as it was not caused by the issue described in this PR, but I let it to your judgement if you want to add it.

Reply:
So, by "plot area width" I'm referring to the pixel width, not a relative width. To me this is less confusing as to what we are attempting to test. Something like this (I think):
```
const plotWidth = plot.width();
const minTickLocation = plot.p2c(ticks[0]);
expect(minTickLocation).toBeLessThan(plot.offset() + (plotWidth / 4));
etc...
```

I don't feel _too_ strongly about this, but to me it's easier to understand what's being tested. With the approach you currently have, I would recommend adding some comments as to what the expect means. Or, perhaps just changing the test name to `creates time ticks in expected locations` would help?

Reply:
Okay, I am pushing now the updated test. Added check for plot area width, with a few minor adjustements:
- Using `ticks[1]` rather than `ticks[0]` as `ticks[0]` is always the left edge thus correctly located, the first generated tick is `ticks[1]` and this was the one incorrect prior to this PR.
- Used es5 `var` to stay consistent with the rest of the file

In summary here is the test now:
```javascript
    it('creates time ticks in expected locations', function() {
        plot = createPlotWithAbsoluteTimeAxis(placeholder, [[[0.95, 1], [1.95, 2]]], undefined, 'seconds');

        var ticks = plot.getAxes().xaxis.ticks;
        expect(ticks.length).toEqual(14);

        // Check that the first generated tick is less than 1/4 of the way along the plot area width
        var firstTickLocation = plot.getAxes().xaxis.p2c(ticks[1].v);
        expect(firstTickLocation).toBeLessThan(plot.offset().left + (plot.width() / 4));
    });
```

Both expects fail before the PR and pass after. What do you think?